include/sound/soc-of.h

```diff
@@ -0,0 +1,22 @@
+/* SPDX-License-Identifier: GPL-2.0
+ *
+ * Copyright 2019 NXP
+ */
+
+#ifndef __LINUX_SND_SOC_OF_H
+#define __LINUX_SND_SOC_OF_H
+
+#include <linux/of.h>
+
+/**
+ * snd_soc_of_mach: DT-based machine driver descriptor
+ *
+ * @drv_name: machine driver name
+ * @of: DT node providing machine driver description
+ */
+struct snd_soc_of_mach {
+	const char *drv_name;
+	struct device_node *of;
```

Comment:
we should really drop the soc_acpi stuff and call it soc_firmware or something.
I don't mind if we do this first, but I think we should avoid using two structures when we can use one...